cpp/arcticdb/storage/azure/azure_storage-inl.hpp

```diff
@@ -293,32 +294,81 @@ bool do_key_exists_impl(
 }
 } //namespace detail
 
+struct CheckAccessibilityTask : async::BaseTask {
+    // This ref assumes the `AzureStorage` will out live the TaskScheduler
```

Comment:
I don't get the comment. Why doesn't the shared_ptr extend the life of the storage?

Reply:
The ordering is controlled here: https://github.com/man-group/ArcticDB/blob/002fa24473cbf222825c65fb6c08c6d8904b1669/cpp/arcticdb/util/global_lifetimes.cpp#L62, where the task_scheduler is deinitialized (waiting on all outstanding threads) before the central shared_ptr to the ModuleData is freed (although as some output frames may also have handles it might not happen at this point. Does the Azure storage need to be shut down here? : https://github.com/man-group/ArcticDB/blob/master/cpp/arcticdb/util/global_lifetimes.cpp#L34

Reply:
The comment is a leftover from a previous iteration.

The field is made `shared_ptr` because the life cycle of the `Library -> Storages -> Storage` is controlled by Python.